pkg/collector/collector.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+	"time"
+
+	eraserv1alpha1 "github.com/Azure/eraser/api/v1alpha1"
+	"github.com/Azure/eraser/pkg/logger"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	pb "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	util "github.com/Azure/eraser/pkg/utils"
+)
+
+const (
+	apiPath = "apis/eraser.sh/v1alpha1"
+)
+
+var (
+	// Timeout  of connecting to server (default: 5m).
+	timeout = 5 * time.Minute
+	log     = logf.Log.WithName("collector")
+)
+
+type client struct {
+	images  pb.ImageServiceClient
+	runtime pb.RuntimeServiceClient
+}
+
+type Client interface {
+	listImages(context.Context) ([]*pb.Image, error)
+}
+
+func (c *client) listImages(ctx context.Context) (list []*pb.Image, err error) {
```

Comment:
looks like `listImages` is shared too?

Reply:
I think we can't take the method away since the Client interfaces for eraser and collector are different. Would it be better to add a listImages to utils and call it from inside listImages from both collector/eraser?